swap_meet/vendor.py

```diff
@@ -1,2 +1,84 @@
 class Vendor:
-    pass
\ No newline at end of file
+    # Set up inventory
+    def __init__(self, inventory = None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+        
+    # Adds an item to the inventory
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    # Removes an item from the vendors inventory
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        if item in self.inventory:
+            self.inventory.remove(item)
```

Comment:
When we write class methods like this, we can separate out what I like to refer to as our nominal case and our edge case. If we take care of the edge case up top in a guard conditional, we don't have to worry about keeping the nominal case in an else statement or its own conditional. 

Our "nominal case" refers to "what should the function do without any funny business". In this method, the nominal case would simply be, it finds the item and removes it. For these nominal cases, we want to try and keep them out of conditionals if at all possible! 

For this specific example, we have already checked the edge case of the item not existing in collection. As a result, we know for a fact that it is in the inventory so we don't need a second conditional to check for that! We can go ahead and just leave the remove statement in line with the guard clause:

```suggestion
        self.inventory.remove(item)
```

Reply:
I really appreciate the idea of separating the edge case and nominal case using a guard clause—it makes the method much more readable and focused. I’m starting to notice how often conditional nesting can get in the way of clarity, and this feels like a great step toward writing cleaner, more professional code.

I’ll definitely try to apply this mindset more consistently moving forward!